cli/medperf/commands/compatibility_test/utils.py

```diff
@@ -138,23 +138,23 @@ def create_test_dataset(
     # TODO: existing dataset could make problems
     # make some changes since this is a test dataset
     config.tmp_paths.remove(data_creation.dataset.path)
-    data_creation.dataset.write()
```

Comment:
In oldish code we were writing dataset multiple times while now we write it only at the very end if hash differs (?).
I don't know if it was made consciously or accidentally; It seems to me new code is working properly also, but I'm not 100% sure we didn't break some legacy crutch that was located there. 
So, I'm just highlighting the change: if you don't expect it, can you look plz one more time to assure it doesn't break anything